pkg/operator/ceph/cluster/controller.go

```diff
@@ -259,6 +264,12 @@ func (r *ReconcileCephCluster) reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, cephCluster, errors.Wrapf(err, "failed to reconcile cluster %q", cephCluster.Name)
 	}
 
+	// update ObservedGeneration in status at the end of reconcile
+	cephCluster.Status.ObservedGeneration = observedGeneration
+	if err := reporting.UpdateStatus(r.client, cephCluster); err != nil {
```

Comment:
I just tried out the branch and didn't see the observedGeneration set on the cephcluster status. I saw it on other CRs such as the pool and object store, just not the cephcluster.

Reply:
The deployment gets updated with the generation and observed generation with every reconcile, not the CRD.

Only for cephcluster it's failing with ```ceph-cluster-controller: failed to set cephcluster "rook-ceph" status. failed to update object "rook-ceph/rook-ceph" status: Operation cannot be fulfilled on cephclusters.ceph.rook.io "rook-ceph": the object has been modified; please apply your changes to the latest version and try again``` need to fix, thanks for pointing.

Reply:
@travisn 
```kubectl get deployment -nrook-ceph  rook-ceph-mds-myfs-a  -o yaml | grep generation
  generation: 1
```
```
kubectl get deployment -nrook-ceph  rook-ceph-mds-myfs-a  -o yaml | grep obser
        f:observedGeneration: {}
  observedGeneration: 1

```


And for updating any changes in the CR,  the CRD `generation` field doesn't get updated. 

I have tried changing the mons count in the cephcluster CR, the CRD generation remains to one but 
```kubectl get crds -nrook-ceph  cephclusters.ceph.rook.io  -o yaml | grep generation
  generation: 1
```

Reply:
@parth-gr The deployment observed generation is not touched by rook, it's just the k8s deployment controller. And instead of querying the generation on the CRD, query it on the CR. It's a suble difference:
`kubectl -n rook-ceph get cephcluster -o yaml | grep generation`

Reply:
Updated now everything works fine.